src/sprite.rs

```diff
@@ -140,6 +140,32 @@ extern "C" fn get_sprite_collision_response(
 }
 
 impl SpriteInner {
+    pub fn get_userdata<T>(&mut self) -> Result<Rc<Box<T>>, Error>
```

Comment:
Why does `get_userdata()` require an exclusive reference here? There must be some invariant that I am unaware of (and which is undocumented in this PR).

Reply:
Ah no reason at all, good catch, that was probably just me blindly copying from set_userdata